### ISSUE TITLE
Control video tag/videojs with wavesurfer?

### ISSUE THREAD

**Issue by chinshr** [katspaugh/wavesurfer.js#698]
Is it possible to create a video tag created by your logic and then have it controlled by wavesurfer? I want to playback HLS content with wavesurfer.

**Comment by katspaugh** [comment]
It's the `mediaType: 'video'` parameter. I've added docs for it on the website.

**Comment by chinshr** [comment]
How would one specify the video tag in the options? In the docs you are specifying to create an audio/video tag, but what if it already exists?

**Comment by katspaugh** [comment]
Oh, an existing video element... Maybe https://github.com/collab-project/videojs-wavesurfer can give you some hints?

@thijstriemstra created it, so maybe he can advice.

**Comment by chinshr** [comment]
Got it, thanks. I will check @thijstriemstra
<https://github.com/thijstriemstra> code.

**Comment by therewasaguy** [comment]
I'm also interested in using wavesurfer with HLS, and I've been working on this in a fork [here](https://github.com/therewasaguy/wavesurfer.js/tree/load-existing-media-elt).

I needed the ability to load an existing audio or video element. So I added a `loadExistingMediaElement` method to wavesurfer.js, and a loadElt method to mediaelement.js.

To use this in my project, I init wavesurfer with `backend: 'MediaElement' `. I have not gotten HLS to work with Web Audio to render the waveform, so I pre-analyze the waveform data, use waveform.util.wjax to get the waveform data, then loadExistingMediaElement like this:

```
  wavesurfer.util.ajax({
    responseType: 'json',
    url: waveformPath // path to waveform.json data
  }).
  on('success', function (data) {
      wavesurfer.loadExistingMediaElement( htmlElement, data);
      wavesurfer.zoom(4);
  });
```

Here are the changes I made so far https://github.com/therewasaguy/wavesurfer.js/commit/49717b5dafb5e6cfd84eb0bd81ee95cb45bec85c

I wonder if this is a good approach? If it's helpful I can clean it up and send a PR. But maybe there's a better way, and I'd be very open to suggestions!

**Comment by katspaugh** [comment]
@therewasaguy sounds like a good approach indeed. A PR would be awesome!

**Comment by chinshr** [comment]
@therewasaguy do you mind creating a PR?

**Comment by therewasaguy** [comment]
@katspaugh @chinshr here's a PR https://github.com/katspaugh/wavesurfer.js/pull/707

**Comment by chinshr** [comment]
@therewasaguy awesome! Thanks so much!

**Comment by chinshr** [comment]
Can this be closed due to #707?
